plugins/RoundRobinGenerator.py

```diff
@@ -10,19 +10,24 @@
 from itertools import cycle
 from typing import Any
 
-from glideinwms.lib.generators import export_generator, Generator, GeneratorError
+from glideinwms.lib.generators import export_generator, Generator
 
 
 class RoundRobinGenerator(Generator[Any]):
     """Round-robin generator"""
 
-    def __init__(self, context: Any = None):
-        super().__init__(context)
-        if "items" not in self.context:
-            raise GeneratorError("items not found in context for RoundRobinGenerator")
-        self.items_cycle = cycle(self.context["items"])
+    items_cycle = None
```

Comment:
Same here. items_cycle should be in the object and not class to allow multiple independent RoundRobinGenerators.
It is OK to leave it None in __init__ and initialize it in generate() 
You can also not define it (directly in generate). Will get a warning but will work. But if there is a class definition you'll have only one